apps/api/pages/api/teams/_post.ts

```diff
@@ -87,24 +124,78 @@ async function postHandler(req: NextApiRequest) {
     ...data,
     metadata: data.metadata === null ? {} : data.metadata || undefined,
   };
-  const team = await prisma.team.create({
+  const awaitingPaymentTeam = await prisma.awaitingPaymentTeam.create({
```

Comment:
Self review: create "AwaitingPaymentTeam" to persist team data submitted to API. "AwaitingPaymentTeam" id will be passed to metadata of stripe checkout session, and after successful payment the id will be used to fetch the corresponding "AwaitingPaymentTeam", create a new "Team" using it, delete "AwaitingPaymentTeam" entry and return the actual "Team" to user.

Reply:
We can't use stripe checkout metadata to store team data, because checkout metadata does not accept keys with values that are booleans, but we allow submitting properties to "/teams", like "isPrivate" that are boolean.